### DISCUSSION TITLE
Global context variables on Nitro

### DISCUSSION THREAD

**Discussion by vflam** [nuxt/nuxt#15947] [Questions]
Hello,

When using Nuxt2+Express in a middleware I could save some data accessible from all API routes, like a variable for DB access and some data that is loaded only once when the server starts using app.set and app.get

Is there something equivalent in Nitro ?

In server/api/myapi.ts:
`export default defineEventHandler((event) => {});`

Is there anywhere in the event variable where I can store data that would be available in all APIs?

Thank you for reading.

**Comment by fathi91** [comment]
Good day,
you could add server middleware in `server/middleware/yourfile.ts` to add data to the request that would be available in any route
```js
export default defineEventHandler((event) => {
      //you can add any data in the context variable
       event.context.userData = {}
});
```

and then you can read it in the requested route

**Comment by puzzle-it-nu** [comment]
This is my server/plugin/ps.ts which initializes knex with Postgresql:
```js
import knex from "knex";

export default defineNitroPlugin(async (nitroApp) => {
  try {
    const pg = knex({
      client: "pg",
      connection: process.env.PG_URL,
      searchPath: [`${process.env.PG_SCHEMA}`, "public"],
      debug: true,
      pool: {
        min: 0,
        max: 10,
      },
    });
    Object.assign(nitroApp, { pg });
    console.log("PostgresDb connection established.");
  } catch (error) {
    console.log("Postgres not available");
  }
});
```

And then where ever I need access to pg I'll do something like this:
```js
    const nitroApp = useNitroApp();
    const { pg } = nitroApp;
    return await pg.select().table(table).debug(false);
```

**Reply by mrguamos** [reply]
can't find any documentation regarding useNitroApp

**Reply by puzzle-it-nu** [reply]
> can't find any documentation regarding useNitroApp
I know this isn't much: https://nitro.unjs.io/guide/advanced/plugins

There is another example, in Discussion too I think, that does something similar but with PrismaClient on the h3 context, I think
